starter-code/src/components/IronContacts.js

```diff
@@ -0,0 +1,107 @@
+import React, {Component} from 'react'
+import contacts from '../contacts.json';
+import '../index.css'
+
+
+class IronContacts extends Component {
+
+  state = {
+    contacts : 5, 
+    datos : contacts.slice(0, 5)
+  }
+
+  addRandomContact = () => {
+    const random = Math.floor(Math.random() * contacts.length-1);
+    //console.log(random)
+    this.setState({
+      contacts : this.state.contacts + 1,
+      datos : [
+        ...this.state.datos,
+        contacts[random]
+      ]
+    })
+  }
+
+  sortByPopularity = () => {
+    const sorted = this.state.datos.sort((a, b) => {
+      const popA=a.popularity
+      const popB=b.popularity
+
+      if (popA < popB) //sort string ascending
+       return 1;
+      if (popA > popB)
+       return -1;
+      return 0; //default return value (no sorting)
+     });
+     this.setState({
+      datos : sorted
+    })
```

Comment:
Te propongo cambiarlo por eso, mira a ver que te parece:
`showContacts: this.state.showContacts.sort(function(a, b){
          return b.popularity-a.popularity
        })`